qt/python/mantidqt/mantidqt/widgets/sliceviewer/models/model.py

```diff
@@ -576,13 +596,27 @@ def _cut_names(self, cut: str):
 
         return xcut_name, ycut_name, help_msg
 
-    def get_hkl_from_xyz(self, xdim, ydim, zdim, xdata, ydata, zdata):
+    @staticmethod
+    def get_dim_indices(slice_point: List[float], qdims: List[int], transpose: bool) -> Tuple[int]:
+        """Returns the indices of the selected x and y axes. It also returns the index of the first non x or y axis."""
+        xdim, ydim = WorkspaceInfo.display_indices(slice_point, transpose)
+        # Get the index of the first axis which is not x or y. Here it is called z.
+        zdim = next(i for i, v in enumerate(slice_point) if v is not None and i in qdims)
```

Comment:
Not sure this is working as expected - this assumes there is only one Q-dim with a slicepoint (see screenshot). 
`xdim` and `ydim` may be non-Q dimensions as well - but I think that's handled OK...